[IMP] PLM: Bring engineering change orders current (#18584)
odoo/documentation: content/applications/inventory_and_mrp/plm/manage_changes/engineering_change_orders.rst

.. |ECO| replace:: :abbr:`ECO (Engineering Change Order)`
.. |ECOs| replace:: :abbr:`ECOs (Engineering Change Orders)`

Use *engineering change orders* (ECOs) to track, implement, and revert change versions made to
products and :doc:`bills of materials <../../manufacturing/basic_setup/bill_configuration>`.

Engineering change orders can be created in three ways:

#. :ref:`Directly in the ECO type <plm/engineering_change_orders/create-eco>`.
#. :ref:`Automatically during an update instructions request in Shop Floor
   <plm/engineering_change_orders/shop-floor>`

> #. Automatically from feedback submitted to the :ref:`ECO type's email alias

Reviewer: Since the other two items in the list have the link on the entire line, I would either change this one to match, or update the others (personally I prefer including less text in the link itself, but I think it is more important to be consistent, so I will leave that up to you!)

   <plm/eco_type/create-type>`.

Create ECO
==========

Engineers and Shop Floor operators can suggest changes in two ways.

.. _plm/engineering_change_orders/create-eco:

Create ECO in PLM
-----------------

To create a new |ECO|, begin by navigating to the **PLM** app. Click the :guilabel:`# Engineering

> Changes` button on the |ECO| type that should be used to track the progress of the change. On the

Reviewer: If there aren't any existing changes, how do they get to the page?

Author: Was there not a **0 Engineering Changes** button on that card? That also gets the user to the Kanban view of the changes.

On the |ECO| form, fill in the following fields accordingly:

- :guilabel:`Description`: Enter a brief summary of the improvement.
- :guilabel:`Type`: Specify the |ECO| type project for organizing the |ECOs|.
- :guilabel:`Apply on`: Specify if the |ECO| changes the version of the :guilabel:`Bill of
  Materials` or the :guilabel:`Product Only`.
- :guilabel:`Product`: Indicate the product being improved.
- :guilabel:`Bill of Materials`: Specify the changed |BOM|. This field auto-populates if the product
  in the :guilabel:`Product` field has an existing |BOM|. If multiple |BOMs| exist, select the
  intended |BOM| from the drop-down menu.

   .. important::
      A :guilabel:`Product` must be selected before the :guilabel:`Bill of Materials` options are
      available.

- :guilabel:`Company`: In multi-company databases, specify if the change applies to products in a
  specific company, or leave blank if the change applies to all companies.

   .. note::

>       The :guilabel:`Company` field is only available to specify with multiple companies enabled.

Reviewer: ```suggestion
      The :guilabel:`Company` field is only available when multiple companies are enabled.
```

#. The :icon:`fa-file-text-o` :guilabel:`Documents` smart button appears, storing the relevant files
   of the |BOM|.
#. A copy of the production |BOM| is stored in the :icon:`fa-flask` :guilabel:`Revision` smart
   button of the |ECO|. The next available version number (e.g., `V2`, `V3`, ...) is also assigned
   to keep track of all |BOM| versions.
#. The :icon:`fa-exchange` :guilabel:`Review Changes` button appears, where all changes made to the
   |ECO| can be reviewed.
#. The stages of the |ECO| :guilabel:`Type` display at the top of the |ECO|.

.. note::
   The :guilabel:`Revision` smart button is available **only** when the :guilabel:`Bill of
   Materials` radio button is selected in the :guilabel:`Apply on` field and the :guilabel:`Start
   Revision` button has been pressed.

.. image:: engineering_change_orders/eco-form.png
   :alt: Started ECO with smart buttons and stages.

.. _plm/engineering_change_orders/shop-floor:

Create ECO in Shop Floor
------------------------

|ECOs| can also be created from the *Shop Floor* module during the manufacturing process.

Open the :menuselection:`Shop Floor` module. Open a work order from the *Overview* or a work center
view.

> In the work order card, click the :icon:`fa-ellipsis-v` :guilabel:`(vertical ellipsis)` button in

Reviewer: I wasn't able to see the ellipsis on the wo card from Shop Floor

Author: Weird. I wonder if you had an operator logged in? I was able to see the ellipsis on my end.

- :guilabel:`Quantity`: Displays the quantity and unit of measure being changed.

The list of :guilabel:`Operations` changes appears after the component changes. The name of the
operation is listed for every operation change. If instructions are updated as part of the change,
its name and type is included on its own line. The :guilabel:`Duration` change is also listed.

Finally, the total cost change appears at the bottom of the report. If total cost change is less
than the previous |BOM|, the cost change appears in **green**. If the total cost change is more than
the previous |BOM|, the cost change appears in **red**.

.. example::
   View the summary of the differences between the current and revised keyboard |BOMs| in the *ECO
   Changes Summary* report of the |ECO|.

   .. image:: engineering_change_orders/bom-changes.png
      :alt: View summary of component changes in the BoM Changes tab.

.. _plm/engineering_change_orders/apply-changes:

Apply changes
=============

After verifying the changes, move the |ECO| to a :ref:`verification stage
<plm/eco_type/stage-config>`, which are stages that require approval before the revised changes can
be applied to the production |BOM|.

> Approvers can :guilabel:`Approve` or :guilabel:`Reject` the changes.

Reviewer: I didn't have the option to "Approve" or "Reject". I moved the eco to in progress and validated, and I only got the "Apply Changes" button.

Author: That was my fault; it was slightly misconfigured -- I just needed to add an approver to make it available on the "In Progress" stage. It is configured properly now if you want to take a look.

(There were tiny little tweaks that I was trying to remember when I was setting this up, so sorry that I missed that!)

.. example::
   After applying the changes of the |ECO| for the :ref:`table
   <plm/engineering_change_orders/example>`, view the version of the current table |BOM| in the
   :guilabel:`Miscellaneous` tab. Here, the :guilabel:`Version` number has been updated to `2`,
   matching the `V2` that appears in the :guilabel:`Revision` smart button of the |ECO|.

   .. image:: engineering_change_orders/bom-version.png
      :alt: View current BOM version in the Miscellaneous tab.

.. _plm/engineering_change_orders/view-eco:

View changes
============

To review proposed changes, navigate to the :menuselection:`PLM app --> Overview`. In the |ECO| type
card, the :guilabel:`# Engineering Changes` button represents the number of operational changes
created.

Click on the :guilabel:`# Engineering Changes` button to open the Kanban view of the |ECO| type. To
view the suggestion, select an |ECO| in the `New` stage.

On the |ECO|, view a summary of the proposed changes in two ways:

- From the :icon:`fa-flask` :guilabel:`Revision` smart button
- From the :icon:`fa-exchange` :guilabel:`Review Changes` smart button

.. example::

>    An operator suggested to verify the angle of the table legs after they had been installed by

Reviewer: ```suggestion
   An operator suggested verifying the angle of the table legs after they had been installed by
```
